geo-types/src/geometry/line_string.rs

```diff
@@ -261,6 +254,37 @@ impl<T: CoordNum> LineString<T> {
         })
     }
 
+    /// Return an iterator yielding one [`Line`] for each line segment in the [`LineString`],
+    /// starting from the **end** point of the LineString, working towards the start.
+    ///
+    /// Note: This is like [`Self::lines`], but the sequence **and** the orientation of
+    /// segments are reversed.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo_types::{wkt, Line, LineString};
+    ///
+    /// let line_string = wkt!(LINESTRING(0 0,5 0,7 9));
+    /// let mut lines = line_string.rev_lines();
+    ///
+    /// assert_eq!(
+    ///     Some(Line::new((7, 9), (5, 0))),
+    ///     lines.next()
+    /// );
+    /// assert_eq!(
+    ///     Some(Line::new((5, 0), (0, 0))),
+    ///     lines.next()
+    /// );
+    /// assert!(lines.next().is_none());
+    /// ```
+    pub fn rev_lines(&'_ self) -> impl ExactSizeIterator<Item = Line<T>> + '_ {
+        self.0.windows(2).rev().map(|w| {
```

Comment:
At first I assumed I wanted to do `line_string.lines().rev()` but that's not quite what I needed.

Given points: `a, b, c`

```
lines():
1. a->b
2. b->c
```

lines().rev():
```
1. b->c
2. a->b
```

But what I want is:
```
rev_lines()
1. c->b
2. b->a
```